sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -1831,8 +1841,8 @@ object PushPredicateThroughNonJoin extends Rule[LogicalPlan] with PredicateHelpe
       }
 
       if (pushDown.nonEmpty) {
-        val pushDownPredicate = pushDown.reduce(And)
-        val replaced = replaceAlias(pushDownPredicate, aliasMap)
+        val replacedByWith = rewriteConditionByWith(pushDown.reduce(And), aliasMap)
+        val replaced = replaceAlias(replacedByWith, aliasMap)
```

Comment:
doesn't `rewriteConditionByWith` already replace the aliases?

Reply:
No, `rewriteConditionByWith` only rewrite common attribute to common expression ref.